src/gleam/base.gleam

```diff
@@ -0,0 +1,41 @@
+import gleam/bit_string.{BitString}
+import gleam/string
+
+external fn erl_encode64(BitString) -> String =
+  "base64" "encode"
+
+external fn erl_decode64(String) -> Result(BitString, Nil) =
+  "gleam_stdlib" "base_decoded4"
+
+/// Encodes a BitString into a base 64 encoded string.
+pub fn encode64(input: BitString, padding: Bool) -> String {
+  let encoded = erl_encode64(input)
+  case padding {
+    True -> encoded
+    False -> string.replace(encoded, "=", "")
+  }
+}
+
+/// Decodes a base 64 encoded string into a BitString.
+pub fn decode64(encoded: String) -> Result(BitString, Nil) {
+  let padded = case bit_string.byte_size(bit_string.from_string(encoded)) % 4 {
+    0 -> encoded
+    n -> string.append(encoded, string.repeat("=", 4 - n))
+  }
+  erl_decode64(padded)
+}
+
+/// Encodes a BitString into a base 64 encoded string with URL and filename safe alphabet.
+pub fn url_encode64(input: BitString, padding: Bool) -> String {
+  encode64(input, padding)
+  |> string.replace("+", "-")
```

Comment:
This might not be the most efficient way of doing this but it was the easiest.
I looked into how the Elixir module defines several encoding alphabets and uses them to generate functions however that is macro heavy.

This library takes the approach of using the build in base64 module then replacing charachters as appropriate https://github.com/potatosalad/erlang-base64url/blob/master/src/base64url.erl#L50